applications/services/desktop/scenes/desktop_scene_pin_input.c

```diff
@@ -98,7 +110,7 @@ bool desktop_scene_pin_input_on_event(void* context, SceneManagerEvent event) {
     Desktop* desktop = (Desktop*)context;
     bool consumed = false;
     uint32_t pin_timeout = 0;
-
+    DesktopScenePinInputState* state = malloc(sizeof(DesktopScenePinInputState));
```

Comment:
You shouldn't need to reallocate a new state here. Instead, retrieve the state that was originally created in the on_enter method, by using scene_manager_get_scene_state function. This will return a ptr to the originally allocated struct, and thus you won't need to reallocate the enter_pin_string below.